File: App/Services/Microservices.Demo.Product.Service/Domain/Products/Entities/Choice.cs
namespace Microservices.Demo.Products.Service.Domain.Products.Entities;

public class Choice
{
    public Choice()
    {
    }

    public Choice(string code, string label)
    {
        Code = code;
        Label = label;
    }

    public string Code { get; }
    public string Label { get; }

    public ChoiceQuestion Question { get; private set; }
}

File: App/Clients/ng-microservices-demo-client/src/app/domain/products/entities/choice.ts
import { AutoMap } from '@automapper/classes';

export class Choice {
  @AutoMap()
  public code!: string;

  @AutoMap()
  public label!: string;

  constructor(
    code: string,
    label: string
  ) {
    this.code = code;
    this.label = label;
  }
}
